pkg/bindinfo/binding_plan_generation.go

```diff
@@ -114,6 +125,15 @@ type state struct {
 // Encode encodes the state into a string.
 func (s *state) Encode() string {
 	sb := new(strings.Builder)
+	for _, t := range s.leading2 {
```

Comment:
encode string will be a burden here, since it's all about string concat.

Reply:
Yes, but it should be OK, since "explain explore" is a lowly-frequency operation here, it can only be triggered by users manually. We want to keep our code as simple as possible at the beginning, and we can optimize this part later when there are real performance problems.